engine/src/main/java/org/terasology/rendering/opengl/FrameBuffersManager.java

```diff
@@ -1,490 +0,0 @@
-     *
-     * @param postProcessor a PostProcessor instance
-     */
-    public void setPostProcessor(PostProcessor postProcessor) {
-        this.postProcessor = postProcessor;
-    }
-
-    /**
-     * TODO: add javadocs
-     *
-     * @param subscriber
-     */
-    public boolean subscribe(FBOManagerSubscriber subscriber) {
-        return fboManagerSubscribers.add(subscriber);
-    }
-
-    /**
-     * TODO: add javadocs
-     *
-     * @param subscriber
-     * @return
-     */
-    public boolean unsubscribe(FBOManagerSubscriber subscriber) {
-        return fboManagerSubscribers.remove(subscriber);
-    }
-
-    public void generateFBOWith(FBOConfig fboConfig) {
-        if (fboUsageCountMap.containsKey(fboConfig.getTitle())) {
-            throw new IllegalArgumentException("There is already an FBO inside FrameBuffersManager, named as: " + fboConfig.getTitle());
-        }
-
-        if (fboConfig.isFBODynamic()) {
-            dynamicFBOConfigs.put(fboConfig.getTitle(), fboConfig);
-        } else {
-            staticFBOConfigs.put(fboConfig.getTitle(), fboConfig);
-        }
-        increaseUsage(fboConfig.getTitle());
-
-        buildFBO(fboConfig);
-    }
-
-    private void increaseUsage(String title) {
-        if (!fboUsageCountMap.containsKey(title)) {
-            fboUsageCountMap.put(title, 1);
-            return;
-        }
-        int usageCount = fboUsageCountMap.get(title) + 1;
-        fboUsageCountMap.put(title, usageCount);
-    }
-
-    /**
-     * TODO: add javadoc
-     *
-     * @param title
-     */
-    public void removeUsage(String title) {
-        Preconditions.checkArgument(fboUsageCountMap.containsKey(title), "The given fbo is not used.");
-
-        if (fboUsageCountMap.get(title) == 1) {
-            getFBO(title).dispose();
-            fboLookup.remove(title);
-            if (dynamicFBOConfigs.containsKey(title)) {
-                dynamicFBOConfigs.remove(title);
-            } else if (staticFBOConfigs.containsKey(title)) {
-                staticFBOConfigs.remove(title);
-            }
-        } else {
-            int usageCount = fboUsageCountMap.get(title);
-            fboUsageCountMap.put(title, usageCount - 1);
-        }
-    }
-
-    private void buildFBO(FBOConfig fboConfig) {
-        String title = fboConfig.getTitle();
-        if (isFBOAvailable(title)) {
-            getFBO(title).dispose();
-            fboLookup.remove(title);
-            logger.warn("FBO " + title + " has been overwritten. Ideally it would have been deleted first.");
-        }
-
-        FBO generatedFBO = fboConfig.generate(fullScale);
-        fboLookup.put(title, generatedFBO);
-    }
-
-    private boolean isFBOAvailable(String title) {
-        return fboLookup.containsKey(title);
-    }
-
-    public boolean isFBOConfigAvailable(String title) {
-        return dynamicFBOConfigs.containsKey(title) || staticFBOConfigs.containsKey(title);
-    }
-
-
-}
-
```

Comment:
Whoa. If you knew how much time it took me to extract this class from the jumble the rendering engine previously was... you'd shed some tears as I'm doing now, sniff sniff... :smile: